emiproc/inventories/swiss.py

```diff
@@ -114,12 +130,37 @@ def __init__(
             for sub in totals.index:
                 catsub = cat + "_" + sub
 
-                if emissions.loc[catsub] < totals[sub]:
+                if cat not in point_source_correction:
                     raise ValueError(
-                        f"Total emissions for {cat=} and {sub=} are negative."
-                        f" point sources: {totals[sub]}, inventory total: {emissions.loc[catsub]}"
+                        f"Category {cat} with point source emissions not in point_source_correction dictionary."
                     )
-                emissions.loc[catsub] -= totals[sub]
+
+                correction = point_source_correction[cat]
+
+                if correction == PointSourceCorrection.KEEP_RASTER_ONLY:
+                    # Remove the emissions of point sources
+                    gdf[sub] = 0.0
+                elif (
+                    correction
+                    == PointSourceCorrection.KEEP_POINT_SOURCE_ONLY_SCALED_TO_RASTER_TOTAL
+                ):
+                    # Scale the point sources to match
+                    total = emissions.loc[catsub]
+                    gdf[sub] = gdf[sub] / totals[sub] * total
+                    # Remove the emissions of the raster
+                    emissions.loc[catsub] = 0.0
+                elif (
+                    correction
+                    == PointSourceCorrection.REMOVE_POINT_SOURCE_FROM_RASTER_TOTAL
+                ):
+                    if emissions.loc[catsub] < totals[sub]:
+                        raise ValueError(
+                            f"Total emissions for {cat=} and {sub=} are negative."
+                            f" point sources: {totals[sub]}, inventory total: {emissions.loc[catsub]}"
+                        )

```

Comment:
Instead of raising an error, another option would be to give a warning and then set the Raster emissions to zero instead of subtracting PS emissions. 
The motivation for this comes from testing with the 2019 inventory for NH3, where the values for eipro from BAFU and the PRTR values are almost the same (point sources: 166159.3, inventory total: 165756.0).

Reply:
I corrected accordingly to your suggestion in 345365e43df68aafd4278161b9f243a27f98a8d4